src/domain/johtoselvitys/Attachments.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { FileInput, IconTrash, Notification } from 'hds-react';
+import { Box } from '@chakra-ui/react';
+import { useMutation, useQueryClient } from 'react-query';
+import { useFormContext } from 'react-hook-form';
+import Text from '../../common/components/text/Text';
+import useLocale from '../../common/hooks/useLocale';
+import FileList from '../../common/components/fileList/FileList';
+import ConfirmationDialog from '../../common/components/HDSConfirmationDialog/ConfirmationDialog';
+import { ApplicationAttachmentMetadata } from '../application/types/application';
+import { deleteAttachment } from '../application/attatchments';
+import { JohtoselvitysFormValues } from './types';
+import ErrorLoadingText from '../../common/components/errorLoadingText/ErrorLoadingText';
+
+type Props = {
+  existingAttachments: ApplicationAttachmentMetadata[] | undefined;
+  attachmentsLoadError: boolean;
+  newAttachments: File[];
+  onAddAttachments: (files: File[]) => void;
+};
+
+function Attachments({
+  existingAttachments,
+  attachmentsLoadError,
+  newAttachments,
+  onAddAttachments,
+}: Props) {
+  const queryClient = useQueryClient();
+  const { t } = useTranslation();
+  const locale = useLocale();
+  const { getValues } = useFormContext<JohtoselvitysFormValues>();
+
+  const [fileToDelete, setFileToDelete] = useState<ApplicationAttachmentMetadata | null>(null);
+  const [showFileDeleteDialog, setShowFileDeleteDialog] = useState(false);
+  const attachmentDeleteMutation = useMutation(deleteAttachment, {
+    onSuccess() {
```

Comment:
How does this work if the application is already deleted or it is not found (404)?

Reply:
Do you mean application or attachment? If the application is not found, this component is not even rendered so there should be no problem.

Reply:
Sorry, I meant attachment.